src/h5web/dataset-visualizer/DimensionMapper.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import ReactSlider from 'react-slider';
+import { Vis, DimensionMapping } from './models';
+import { useVisProps } from './VisProvider';
+import ButtonGroup from '../visualizations/shared/ButtonGroup';
+import styles from './DimensionMapper.module.css';
+
+interface Props {
+  activeVis: Vis;
+  setMapping: (d: DimensionMapping) => void;
+}
+
+function DimensionMapper(props: Props): JSX.Element {
+  const { activeVis, setMapping } = props;
+  const { rawDims, mapping } = useVisProps();
+
+  if (activeVis === Vis.Line && rawDims.length === 2) {
+    return (
+      <div className={styles.mapper}>
+        <div className={styles.buttonGroupWrapper}>
+          <span className={styles.buttonGroupLabel}>X</span>
+          <ButtonGroup
+            className={styles.buttonGroup}
+            buttonClassName={styles.btn}
+            buttons={Object.keys(rawDims).map(dimKey => {
+              const dimIndex = Number(dimKey);
+              return {
+                label: dimKey,
+                isSelected: mapping.x === dimIndex,
+                onClick: () => {
+                  if (mapping.x !== dimIndex) {
+                    const newMapping: DimensionMapping = {
+                      x: dimIndex,
+                      slicingIndices: { [mapping.x]: 0 },
+                    };
+                    setMapping(newMapping);
```

Comment:
Well except here... the `newMapping` set the new `dimIndex` as `x` and the slicingIndex of the old `x` (in `mapping.x`) back to 0.